### PR TITLE
procedures: rewrite User Guide: Using a Git provider access token

### PR DESCRIPTION
**Pull request by max-cx**
## What does this pull request change?
Rewrite the page https://www.eclipse.org/che/docs/stable/end-user-guide/using-git-credentials/ in response to new info from the SMEs.

## What issues does this pull request fix or reference?
https://issues.redhat.com/browse/RHDEVDOCS-4443

## Specify the version of the product this pull request applies to
`main` (`7.56.x` and `7.57.x`)
`7.58.x`

## Pull Request checklist

The author and the reviewers validate the content of this pull request with the following checklist, in addition to the [automated tests](code_review_checklist.adoc).

- Any procedure:
  - [x] Successfully tested.
- Any page or link rename:
  - [ ] The page contains a redirection for the previous URL.
  - Propagate the URL change in:
    - [ ] Dashboard [default branding data](https://github.com/eclipse-che/che-dashboard/blob/main/packages/dashboard-frontend/src/services/bootstrap/branding.constant.ts)
    - [ ] Chectl [constants.ts](https://github.com/che-incubator/chectl/blob/main/src/constants.ts)
- [ ] Builds on [Eclipse Che hosted by Red Hat](https://workspaces.openshift.com).
- [ ] the *`Validate language on files added or modified`* step reports no vale warnings.

### CODE REVIEW

**Review by nallikaea — APPROVED** [review]

File: modules/end-user-guide/pages/authenticating-to-a-git-server-from-a-workspace.adoc [hunk 1]
```diff
@@ -9,12 +9,12 @@
 
 In a workspace, you can run Git commands that require user authentication like cloning a remote private Git repository or pushing to a remote public or private Git repository.
 
-User authentication to a Git server from a workspace can be configured by the administrator or user:
+User authentication to a Git server from a workspace is configured by the administrator or in some cases by the individual user:
```
Comment (rolfedh) [inline modules/end-user-guide/pages/authenticating-to-a-git-server-from-a-workspace.adoc]:
```suggestion
User authentication to a Git server from a workspace is configured by the administrator or, in some cases, by the individual user:
```

File: modules/end-user-guide/pages/authenticating-to-a-git-server-from-a-workspace.adoc [hunk 2]
```diff
@@ -9,12 +9,12 @@
...
 
 * Your administrator sets up an xref:administration-guide:oauth-for-github-gitlab-or-bitbucket.adoc[OAuth application on GitHub, GitLab, or Bitbucket] for your organization's {prod} instance.
 
-* As a user, you create and apply your own {kubernetes} Secrets for your xref:using-git-credentials.adoc[Git credentials store and access token].
+* As a workaround, users create and apply their own {kubernetes} Secrets for their personal xref:using-a-git-provider-access-token.adoc[Git-provider access tokens].
```
Comment (rolfedh) [inline modules/end-user-guide/pages/authenticating-to-a-git-server-from-a-workspace.adoc]:
```suggestion
* As a workaround, some users create and apply {kubernetes} Secrets as personal xref:using-a-git-provider-access-token.adoc[Git-provider access tokens].
```

File: modules/end-user-guide/pages/using-a-git-provider-access-token.adoc [hunk 3]
```diff
@@ -0,0 +1,118 @@
+:_content-type: PROCEDURE
+:description: Using a Git-provider access token
+:keywords: Git, credentials, access-token
+:navtitle: Using a Git-provider access token
+:page-aliases: using-a-Git-credentials-store.adoc, using-git-credentials.adoc, 
+
+[id="using-a-git-provider-access-token"]
+= Using a Git-provider access token
+
+OAuth for GitHub, GitLab, or Bitbucket needs to be xref:administration-guide:oauth-for-github-gitlab-or-bitbucket.adoc[configured by the administrator] of your organization's {prod-short} instance. If it was not possible for your administrator to configure it for {prod-short} users, the workaround is for you to apply your personal access token as a {kubernetes} Secret.
```
Comment (rolfedh) [inline modules/end-user-guide/pages/using-a-git-provider-access-token.adoc]:
```suggestion
OAuth for GitHub, GitLab, or Bitbucket needs to be xref:administration-guide:oauth-for-github-gitlab-or-bitbucket.adoc[configured by the administrator] of your organization's {prod-short} instance. If your administrator could not configure it for {prod-short} users, the workaround is for you to apply your personal access token as a {kubernetes} Secret.
```

File: modules/end-user-guide/pages/using-a-git-provider-access-token.adoc [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+
+Mounting your access token as a Secret enables the {prod-short} Server to access the remote repository that is cloned during workspace creation, including access to the repository's `/.che` and `/.vscode` folders.
+
+Apply the Secret in your user {orch-namespace} of the {orch-name} cluster of your organization's {prod-short} instance.
+
+After you have applied the Secret, you can create new workspaces from a private GitHub, GitLab, or Bitbucket-server repository.
```
Comment (rolfedh) [inline modules/end-user-guide/pages/using-a-git-provider-access-token.adoc]:
```suggestion
After applying the Secret, you can create new workspaces from a private GitHub, GitLab, or Bitbucket-server repository.
```

**Review by rolfedh — APPROVED** [review]
Nice work, @max-cx. I've offered some suggestions that might be useful.

File: modules/end-user-guide/pages/using-a-git-provider-access-token.adoc [hunk 5]
```diff
@@ -0,0 +1,118 @@
...
+----
+$ oc apply -f - <<EOF
+__<Secret_prepared_in_step_5>__
+EOF
+----
+====
+
+.Verification
+
+. xref:starting-a-new-workspace-with-a-clone-of-a-git-repository.adoc[Start a new workspace by using the URL of a remote Git repository] that is hosted by that Git provider.
```
Comment (rolfedh) [inline modules/end-user-guide/pages/using-a-git-provider-access-token.adoc]:
```suggestion
. xref:starting-a-new-workspace-with-a-clone-of-a-git-repository.adoc[Start a new workspace by using the URL of a remote Git repository] that the Git provider hosts.
```

File: modules/end-user-guide/pages/authenticating-to-a-git-server-from-a-workspace.adoc [hunk 6]
```diff
@@ -9,12 +9,12 @@
 
 In a workspace, you can run Git commands that require user authentication like cloning a remote private Git repository or pushing to a remote public or private Git repository.
 
-User authentication to a Git server from a workspace can be configured by the administrator or user:
+User authentication to a Git server from a workspace is configured by the administrator or, in some cases, by the individual user:
 
 * Your administrator sets up an xref:administration-guide:oauth-for-github-gitlab-or-bitbucket.adoc[OAuth application on GitHub, GitLab, or Bitbucket] for your organization's {prod} instance.
 
-* As a user, you create and apply your own {kubernetes} Secrets for your xref:using-git-credentials.adoc[Git credentials store and access token].
+* As a workaround, users create and apply their own {kubernetes} Secrets for their personal xref:using-a-git-provider-access-token.adoc[Git-provider access tokens].
```
Comment (max-cx) [inline modules/end-user-guide/pages/authenticating-to-a-git-server-from-a-workspace.adoc]:
```suggestion
* As a workaround, some users create and apply their own {kubernetes} Secrets for their personal xref:using-a-git-provider-access-token.adoc[Git-provider access tokens].
```

File: modules/end-user-guide/pages/using-a-git-provider-access-token.adoc [hunk 7]
```diff
@@ -0,0 +1,118 @@
...
+----
+kind: Secret
+apiVersion: v1
+metadata:
+  name: personal-access-token-__<your_chosen_name_for_this_token>__
+  labels:
+    app.kubernetes.io/component: scm-personal-access-token
+    app.kubernetes.io/part-of: che.eclipse.org
+  annotations:
+    che.eclipse.org/che-userid: _<{prod-id-short}_user_ID>_# <1>
```
Comment (max-cx) [inline modules/end-user-guide/pages/using-a-git-provider-access-token.adoc]:
```suggestion
    che.eclipse.org/che-userid: __<{prod-id-short}_user_id>__# <1>
```

File: modules/end-user-guide/pages/using-a-git-provider-access-token.adoc [hunk 8]
```diff
@@ -0,0 +1,118 @@
...
+    che.eclipse.org/scm-personal-access-token-name: _<git_provider_name>_# <2>
+    che.eclipse.org/scm-url: _<Git_provider_endpoint># <3>
```
Comment (max-cx) [inline modules/end-user-guide/pages/using-a-git-provider-access-token.adoc]:
```suggestion
    che.eclipse.org/scm-url: __<git_provider_endpoint>__# <3>
```

File: modules/end-user-guide/pages/using-a-git-provider-access-token.adoc [hunk 9]
```diff
@@ -0,0 +1,118 @@
...
+    che.eclipse.org/scm-userid: '_<Git_provider_user_ID>_'# <4>
```
Comment (max-cx) [inline modules/end-user-guide/pages/using-a-git-provider-access-token.adoc]:
```suggestion
    che.eclipse.org/scm-userid: '__<git_provider_user_id>__'# <4>
```

File: modules/end-user-guide/pages/using-a-git-provider-access-token.adoc [hunk 10]
```diff
@@ -0,0 +1,118 @@
...
+    che.eclipse.org/scm-username: _<Git_provider_username>_
```
Comment (max-cx) [inline modules/end-user-guide/pages/using-a-git-provider-access-token.adoc]:
```suggestion
    che.eclipse.org/scm-username: __<git_provider_username>__
```

File: modules/end-user-guide/pages/using-a-git-provider-access-token.adoc [hunk 11]
```diff
@@ -0,0 +1,118 @@
...
+* Bitbucket Server: link:https://developer.atlassian.com/server/bitbucket/rest/v802/api-group-api/#api-api-latest-users-userslug-get[Get users]. See the `id` value in the response.
+
+. Prepare a new {orch-name} Secret.
++
+[source,yaml,subs="+quotes,+attributes,+macros"]
+----
+kind: Secret
+apiVersion: v1
+metadata:
+  name: personal-access-token-__<your_chosen_name_for_this_token>__
```
Comment (max-cx) [inline modules/end-user-guide/pages/using-a-git-provider-access-token.adoc]:
```suggestion
  name: personal-access-token-__<your_choice_of_name_for_this_token>__
```

**Review by rkratky — APPROVED** [review]

**Pull request merged**